#293 Fix links in "W3C Webdriver Spec" section of the API docs

ParmarKrishna opened the pull request:
@garg3133 Review Requested for Fix Issue #293

As per my recent investigations on #289, I implemented same method of `if-else html escaped rendering` for links, and it solves the issue.

I did:

- Add WEBDRIVER_SPEC URL in `/postdoc.config.js` and export it in appSettings.
- Import appSettings in the `if-else` when link do not have http as protocol

Do let me know if this works, or should I look for other way around.

ParmarKrishna committed:
#293 Fix Webdriver Spec

beatfactor commented on postdoc.config.js:
```diff
@@ -6,7 +6,8 @@ const {
   API_DOCS_FOLDER = resolve('../nightwatch/lib/api'),
-  EXAMPLES_FOLDER = 'node_modules/nightwatch-examples/tests'
+  EXAMPLES_FOLDER = 'node_modules/nightwatch-examples/tests',
+  WEBDRIVER_SPEC = 'https://www.w3.org/TR/webdriver',
```
the correct url is https://w3c.github.io/webdriver/

ParmarKrishna replied:
Sure. Making it right in a moment.

ParmarKrishna replied:
Waiting on #291 Merge status, to update this PR with new commit.

garg3133 replied:
@ParmarKrishna You should be able to work on multiple PRs simultaneously by checking out between the branches.

ParmarKrishna replied:
okay. noted.

garg3133 commented:
> Import appSettings in the if-else when link do not have http as protocol

I do not see any implementation for this. We should only add the WebDriver Spec link as prefix if the already does not start with `http`.

ParmarKrishna commented:
I want to ask,
As can be seen in commit, If I can define the variable(WEBDRIVER_SPEC, exported as webdriverSpec) that specifies link in the `postdoc.config.js` and then reference it with appSettings.webdriverSpec in the `/src/includes/api-method.ejs`
Or should I hardcode the href with link as prefix?

ParmarKrishna committed:
Merge branch 'nightwatchjs:versions/3.0' into #293-Fix-WebDriver-Spec-Links

garg3133 commented:
Your current approach seems good.

ParmarKrishna committed:
Fix correct link address in postdoc.config.js

WEBDRIVER_SPEC previously had wrong link address, after review from @beatfactor , the correct link address has been added.

ParmarKrishna commented:
@garg3133 Review Requested.

I have corrected the link address, changed to [https://w3c.github.io/webdriver](https://w3c.github.io/webdriver/) from [https://www.w3.org/TR/webdriver](https://www.w3.org/TR/webdriver)

ParmarKrishna committed:
Add if-else block in 'src/includes/api-method.ejs'

Adding code if "W3C Webdriver Spec" link do no start with "http", it redirects to the W3C documentation, and if it has "http", it redirects to that specific link.

Addtionally, added class property for the unordered list named, "webdriver-spec-related-links"

garg3133 commented on postdoc.config.js:
```diff
@@ -87,6 +88,7 @@ export default {
   appSettings: {
     version: NIGHTWATCH_VERSION,
     baseUrl: BASE_URL,
+    webdriverSpec : WEBDRIVER_SPEC,
```
```suggestion
    webdriverSpecUrl : WEBDRIVER_SPEC,
```

garg3133 commented on src/includes/api-method.ejs:
```diff
@@ -142,8 +142,12 @@ npx nightwatch <%- method.exampleLink %></code></pre>
-            <li><code><a href="<%= method.link %>" target="_blank"><%- method.link %></a></code></li>
+        <ul class="webdriver-spec-related-links">
+            <% if (method.link.startsWith('http')) { %>
+                <li><code><a href="<%- link %>"><%= link %></a></code></li>
```
Please test your changes properly before asking for review. `link` here is not defined, it should be `method.link` instead.

garg3133 replied:
Plus, these links should also open in a new tab.

garg3133 commented on src/includes/api-method.ejs:
```diff
@@ -142,8 +142,12 @@ npx nightwatch <%- method.exampleLink %></code></pre>
         <h3>W3C WebDriver spec</h3>
-        <ul>
-            <li><code><a href="<%= method.link %>" target="_blank"><%- method.link %></a></code></li>
+        <ul class="webdriver-spec-related-links">
```
```suggestion
        <ul class="webdriver-spec-link">
```

garg3133 commented on src/includes/api-method.ejs:
```diff
@@ -142,8 +142,12 @@ npx nightwatch <%- method.exampleLink %></code></pre>
+            <% if (method.link.startsWith('http')) { %>
+                <li><code><a href="<%- link %>"><%= link %></a></code></li>
+           <% } else { %>
+            <li><code><a href="<%- appSettings.webdriverSpec -%><%= method.link %>" target="_blank"><%- method.link %></a></code></li>
```
HTML escaping should be done in the text that would appear in the document and not in the `href` tag.
```suggestion
            <li><code><a href="<%- appSettings.webdriverSpec -%><%- method.link %>" target="_blank"><%= method.link %></a></code></li>
```

ParmarKrishna committed:
Refine code as per review

ParmarKrishna commented on src/includes/api-method.ejs:
```diff
@@ -142,8 +142,12 @@ npx nightwatch <%- method.exampleLink %></code></pre>
-            <li><code><a href="<%= method.link %>" target="_blank"><%- method.link %></a></code></li>
+        <ul class="webdriver-spec-link">
+            <% if (method.link.startsWith('http')) { %>
+                <li><code><a href="<%- method.link %>" target="_blank"><%= method.link %></a></code></li>
```
Added target="_blank" to open link in new page, even if it starts with http.

ParmarKrishna commented on src/includes/api-method.ejs:
```diff
@@ -142,8 +142,12 @@ npx nightwatch <%- method.exampleLink %></code></pre>
         <h3>W3C WebDriver spec</h3>
-        <ul>
-            <li><code><a href="<%= method.link %>" target="_blank"><%- method.link %></a></code></li>
+        <ul class="webdriver-spec-link">
```
Renamed class name

ParmarKrishna commented:
@garg3133 Review Requested

garg3133 reviewed (approved):
Looks good now, thanks!
